src/LoginForm/LoginForm.js

```diff
@@ -1,32 +1,23 @@
 import React, { Component } from 'react';
-import { NavLink } from 'react-router-dom';
 import './LoginForm.css';
 import Areas from '../Areas/Areas';
+import App from '../App/App';
+
 import map from '../assets/map.png';
 
 class LoginForm extends Component {
-  constructor() {
+  constructor(props) {
     super();
-    this.state = {
-      name: '',
-      email: '',
-      type: ''
-    }
   }
 
   handleChange = (event) => {
     let key = event.target.id || event.currentTarget.value;
-    this.setState({ [key]: event.target.value })
+    this.props.change(key, event);
   }
 
   handleSubmit = (event) => {
     event.preventDefault();
-    if (this.state.name === '' || this.state.email === '' || this.state.type === '') {
```

Comment:
If this logic is removed, how are we preventing a user from logging in with incomplete data?

Reply:
The logic isn't totally removed, just moved over into the app component.

Reply:
Ok, I see it now. I'm not sure how I missed that! Thanks!